client/client-multi/client-c/src/kaa/platform-impl/common/kaa_client_properties.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2014-2016 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <platform/kaa_client_properties.h>
+#include <platform/stdio.h>
+#include <platform/mem.h>
+#include "utilities/kaa_mem.h"
+
+#define KAA_DEFAULT_WORKING_DIRECTORY "\0"
+
+static struct kaa_client_props_t {
+    char* working_directory;
+} kaa_client_props;
+
+kaa_error_t kaa_client_props_set(const kaa_client_props_t *self)
+{
+    if (!self) {
+        kaa_client_props.working_directory = KAA_MALLOC(strlen(KAA_DEFAULT_WORKING_DIRECTORY));
```

Comment:
Uh, oh! You forgot to allocate additional byte for NULL-termination everywhere in your patch :c